index.js

```diff
@@ -1,46 +1,32 @@
 "use strict";
 
 const connoisseur = require("./lib/connoisseur");
+const fs = require("fs");
+const sourceFiles = require("yargs").argv._;
 
-const CODE = `
-"use strict";
-
-// lalala
-[4, 9, 16].map(Math.sqrt);  // → [ 2, 3, 4 ]
-const sth = function () { return "lalala" === 1; };  // asd
-sth();  // eslint-disable-line no-unused-expressions
-// → false
-/* eslint-disable no-unused-expressions */
-"test str";
-"test";  // → 'test'
-/* eslint-enable no-unused-expressions */
-const time = 3 * 15;  // eslint-disable-line no-unused-vars
-// endcomm
-`;
-
-const result = connoisseur(CODE, "stdin");
+const displayErrors = function (errors) {
+    let hasError = false;
+    errors.forEach(file => {
```

Comment:
Using filter first to remove files without errors would make the code more readable. May incur some overhead due to the extra loop but shouldn't be too significant.